### ISSUE TITLE
Unable to set a default font for Icons in theme

### ISSUE THREAD

**Issue by Bendetta** [GeekyAnts/NativeBase#5001]
### Description

I would like to set a default font for icons in my theme so I don't have to specify `as={MaterialIcons}` every time I use an Icon when using only one icon set. Other default properties can be specified but `as` seems to be a special exception for some reason.

### CodeSandbox/Snack link

https://snack.expo.dev/47A_iMdvj

### Steps to reproduce

1. Create a custom theme
2. Set defaultProps for `Icon` to `as: <FontHere>`
3. Render an icon with `<Icon name="<icon name here> />`
4. See icon won't render without specifying `as`

### NativeBase Version

3.4.3

### Platform

- [X] Android
- [ ] CRA
- [ ] Expo
- [X] iOS
- [ ] Next

### Additional Information

You could probably use `resolvedProps.as` instead of [props.as](https://github.com/GeekyAnts/NativeBase/blob/master/src/components/primitives/Icon/Icon.tsx#L15) but I don't know know the implications of doing so.

**Comment by mayank-96** [comment]
Hello @Bendetta, Thanks for reporting this issue, we'll fix it soon!

**Comment by Viraj-10** [comment]
Hi @Bendetta, This issue has been fixed in `v3.4.6`.

**Comment by RobertSasak** [comment]
Thank you for the PR, it works! For those who wonder how to use it here is a small snippet.
```tsx
import { MaterialCommunityIcons } from '@native-base/icons'

const theme = extendTheme({
    components: {
        Icon: {
            defaultProps: {
                as: MaterialCommunityIcons,
            },
        },
    },
})

// Usage
// Notice that there is no need to specify "as" prop
const back = () => <Icon size="lg" name="chevron-left" color="white" />
```
Be aware that this also upsets Typescript and you will get a following Typescript error when trying to set _size_ prop of Icon.
```
Type 'string' is not assignable to type 'ThemeComponentSizeType<"Icon">'.ts(2322)
types.d.ts(18, 5): The expected type comes from property 'size' which is declared here on type 'IntrinsicAttributes & InterfaceIconProps & Partial<{}> & RefAttributes<unknown>'
```
This issue appers regardles of version 3.4.5 or 3.4.6.
